### ISSUE TITLE
Custom Environments?

### ISSUE THREAD

**Issue by MaggieK410** [robocasa/robocasa-gr1-tabletop-tasks#3]
Hi, 

Thank you for this very useful extension! I was wondering if there is any guidance on how custom environments could be integrated into this pipeline. 

I have a modified robosuite Lift environment in which I collected demonstrations and trained GR00T for. I would like to evaluate it now using the simulation, but Im not quite sure how I can transfer the robosuite environment here. Possibly some addition to robocasa/environments ?

Any help is greatly appreciated!

**Comment by MaggieK410** [comment]
Hi @squarefk 
Thank you for the very quick answer and the useful hint, I got it the simulation to work now but still have some questions. As context: I actually wanted to use GR1ArmsOnly but this is not supported yet in the gr1_unified namespace, so now I pass an empty vector for the hip movement. I trained a GR00T arms only fourier hands model for 20 000 steps on the 96 trajectories that I collected and I also trained visual as I've seen people say it works better. Here is one example:

https://github.com/user-attachments/assets/1e307692-2638-447e-8615-e4b7055c020f


The loss at the end of training was ~0.007. I also evaluated it using the Isaac-GR00T/getting_started/2_finetuning.ipynb, where the model seems to work quite well. Here is the plot for the shoulder roll, but all joints look very similar to this:


<img width="1005" height="515" alt="Image" src="https://github.com/user-attachments/assets/2f7ddd68-4ad6-41b0-b64e-f0f15ce77d14" />

However, when deploying the model into the simulation environment, it actually does not perform hat well. Here is an example video collected after I got this script to work:

https://github.com/user-attachments/assets/92f08d19-c699-4a33-acd8-add1d04ad10b

The success rate on 30 episodes is 0%, despite the plots from the 2_finetuning.ipynb looking pretty good. Im a bit confused if this is related to the environment controls, or the finetuning of the model. Do you have any hints on how to improve the model's performance here and why it does not work well? Does it have to do with the simulation envrionment or the training itself? Thank you in advance!

**Comment by 4133223** [comment]
@MaggieK410 If possible, I would like to know the methods you use to manually control the robot in the simulation environment and collect data.

**Comment by squarefk** [comment]
@MaggieK410  Have you successfully tried using an adjusted action horizon? Additionally, you could try fine-tuning with a smaller batch size and fewer training steps.

**Comment by MaggieK410** [comment]
@squarefk I am currently running a model with an action horizon of 25, but I was only able to start train this this morning. For this, I passed action_horizon=25 to the [pre-trained model](https://github.com/NVIDIA/Isaac-GR00T/blob/main/scripts/gr00t_finetune.py#L185). I also had to change the [hardcoded 16 in the dataconfig](https://github.com/NVIDIA/Isaac-GR00T/blob/main/gr00t/experiment/data_config.py#L65). I still have the issue that the action horizon in the action head config is 16, but Im trying to solve that also now. Or did you mean only increasing the action horizon during inference and not during training?

For action horizon 16, I also trained with a batch size of 1 for 20 000 steps and I realized I get better results (30% success rate sometimes) when using earlier checkpoints (e.g. 14 000 or 13 000). I also made a dataset for picking up small spheres, which requires a bit more precision and there I am back to 0% success rate. Increasing the batch size did not improve performance, although that was before merging [](https://github.com/NVIDIA/Isaac-GR00T/pull/264), so now it might be better. 

I will report back as soon as I have new results, and would greatly appreciate any guidance on how to change the action horizon, since what Im doing is a bit hacky and maybe not how its intended.

**Comment by squarefk** [comment]
Closing this issue for now, as there has been no further feedback from the user. Feel free to reopen it if there are any questions in the future.
